iceoryx_dds/include/iceoryx_dds/gateway/channel.hpp

```diff
@@ -62,21 +63,45 @@ class Channel
     Channel(const iox::capro::ServiceDescription& service,
             const IceoryxTerminalPtr iceoryxTerminal,
             const DDSTerminalPtr ddsTerminal) noexcept;
+    Channel(const iox::capro::ServiceDescription& service,
+            const IceoryxTerminalPtr iceoryxTerminal,
+            const DDSTerminalPtr ddsTerminal,
+            const uint64_t& dataSize) noexcept;
+
     constexpr bool operator==(const Channel<IceoryxTerminal, DDSTerminal>& rhs) const noexcept;
 
+    ///
+    /// @brief create Creates a channel for the given service whose terminals reside in a static object pool.
+    /// @param service The service to create the channel for.
+    /// @return A copy of the created channel, if successful.
+    /// @note The channel will be created with an unknown data element size.
```

Comment:
The implementation will now detect the size by peeking at the next available value that has been received.
This means the size does not need to be manually specified anymore.

However, we cannot deal with "dynamic message sizes" since Posh doesn't support this.

Reply:
How you came to this conclusion? You can have another chunk size with each allocateChunk()

Reply:
And if you do a getChunk with the ChunkHeader this should give you the size of the received chunk which can be dynamic as well

Reply:
Allocating a different sized chunk from a publisher does the following:
```
[ Error ]: ICEORYX error! POSH__SENDERPORT_SAMPLE_SIZE_CHANGED_FOR_ACTIVE_PORT
gateway_dds2iceoryx: iceoryx/iceoryx_utils/source/error_handling/error_handling.cpp:56: static void iox::ErrorHandler::ReactOnErrorLevel(iox::ErrorLevel, const char*): Assertion `false' failed.
Aborted (core dumped)
```

Reply:
Ah I just had a look at a code, there is a flag that needs to be set: `useDynamicPayloadSizes`

Reply:
Yes, I'm not that happy with this flag. I guess we will handle this differently with the #25 refactoring